index.rst

```diff
@@ -10,20 +10,357 @@
+This is slightly problematic for client/server, because the journal file will need to be populated with information we get from the Registry database; this means the client cannot be responsible for creating the journal file unless we make fetching the datastore records and deleting them separate operations.
+That isn't too bad - it's just a slight increase in latency and a bit more http traffic.
+Another alternative would be to have the server take responsibility for creating the journal file, and then either returning responsibility for its deletion to the client or taking responsibility for both the deletion of the Datastore artifacts and the deletion of the journal file.
+Which of these is preferable probably depends on whether we want these operations to block until completion and whether we have reasons to perform other Datastore operations on the server (up to this point, having the client use a vanilla ``FileDatastore`` and perform all Datastore operations still seems viable).
+
+.. note::
+   This proposal should not affect our ability support disassembled composites, though we may be able to make further simplifications if we drop that support.
+   It may have implications for our ability to support multiple datasets in a single file, at least in terms of safeguarding against premature deletions
+   of those files.
+   An easy way to mitigate that would be to limit that support to "unmanaged" datasets that are ingested with absolute URIs, since those are never deleted by butler code, and this satisfies Rubin's only current use case (ingesting DECam raws) for this functionality.
+
+.. _including-signed-urls-for-access-control:
+
+Including signed URLs for access control
+----------------------------------------
+
+Our access control model for the official Rubin Observatory data repositories (see :cite:`DMTN-169`) is based on information stored in the Registry - collection names, whenever possible, and new naming conventions or new columns in contexts (e.g. dimension records or dataset type names) that are not associated with collections.
+Access to the associated files managed by a Datastore is mediated by signed URLs; once the server side of the remote Butler has determined that an API call is permitted (based on that the Registry-side information), it can generate one or more signed URLs to pass to Datastore that provide direct access to the controlled files.
+A seemingly natural place to include URL-signing in this proposal is inside the Datastore implementation, since only the Datastore knows where any URLs might exist in the opaque-to-Registry records it uses, and only the Datastore ever uses and kind of URLs.
+This approach has two major drawbacks, however:
+
+- It requires the Datastore to have a server component; in every other respect we can abstract the differences between a SQL-backed full Butler and a client/server full Butler via a different Registry implementation (see :ref:`public-interface-changes`).
```

Comment:
If Datastore is only client-side but is providing records for server-side Registry to store, we need to make sure they're for proper UUIDs (accessible to the user) and we should protect against excessive resource consumption.